### ISSUE TITLE
"Find lost" by last location when connected (for both buds)

### ISSUE THREAD

**Issue by TheLastGimbus** [TheLastGimbus/FreeBuddy#18]
I invented this app when one day I lost one of my 4i's buds... the Pro's have the "find lost" feature... but 4i's don't... while they are perfectly capable of that??? Why??

Yet, almost 3 years passed from that day (I later found the bud :D), and FreeBuddy still can't do that

## Reasons:
The headphones do communicate whether one or two buds are currently worn, either directly or by showing only selective battery values... we can detect when one of them was disconnected.... BUT, only when we have active bluetooth serial communication with them - that means, the app needs to:
### Be constantly active in background
This may result in battery usage + poor UX - this has to be done *right*, and would potentialy require one extra onboarding screen for user to select if they want it... and then, two separate logics for the app to go...

### Use GPS many times
...like, every time you hide one headphone in case, or something. This has to be done even more *right*, because i don't want my battery drained, do you?

### Save those location in a database
This brings a lot of work to code and keep data safe

### ...display them on a map
...so we need an online map provider, and an interacvite map, all done nice-and-pretty - this is *some* work

### Connect to headphones from two places (background and UI) simontaniously
This is not so obvious in Flutter world. Right now, there is a clear `Stream`s and `Futures` pipeline that works great with Flutter widget - we will need some good way to sync all of that up in two places at a time

# TL;DR
It's hard. But worth it. Like literal money for lost buds 👇
